src/main/java/com/researchspace/service/impl/NfsManagerImpl.java

```diff
@@ -52,15 +55,60 @@ public void saveNfsFileStore(NfsFileStore fileStore) {
     nfsDao.saveNfsFileStore(fileStore);
   }
 
+  @Override
+  public NfsFileStore createAndSaveNewFileStore(
+      Long fileSystemId, String fileStoreName, String fileStorePath, User user) {
+
+    NfsFileSystem fileSystem = getFileSystem(fileSystemId);
+    String pathToSave = fileStorePath;
+    String nfsServerUrl = fileSystem.getUrl();
+    int serverUrlIdxInPath = pathToSave.indexOf(nfsServerUrl);
+    int serverUrlLength = nfsServerUrl.length();
+    if (serverUrlIdxInPath >= 0 && pathToSave.length() > serverUrlLength) {
+      pathToSave = pathToSave.substring(serverUrlIdxInPath + serverUrlLength);
+    }
+    pathToSave = NfsFileStore.validateTargetPath(pathToSave);
+
+    NfsFileStore userStore = new NfsFileStore();
+    userStore.setUser(user);
+    userStore.setName(fileStoreName);
+    userStore.setPath(pathToSave);
+    userStore.setFileSystem(fileSystem);
+    saveNfsFileStore(userStore);
+    return userStore;
+  }
+
   @Override
   public void markFileStoreAsDeleted(NfsFileStore fileStore) {
     fileStore.setDeleted(true);
     nfsDao.saveNfsFileStore(fileStore);
   }
 
   @Override
-  public List<NfsFileStore> getFileStoresForUser(Long userId) {
-    return nfsDao.getUserFileStores(userId);
+  public List<NfsFileStoreInfo> getFileStoreInfosForUser(User user) {
```

Comment:
You could return a `Set` here so it will be quicker to check if the file file store name is already exsisting in `verifyFileStoreNameUniqueForUser` 

But you need to change equals/hashcode on `NfsFileStoreInfo` to be based the `name` rather than `id`

Reply:
Yeah the filestore name validation currently takes a bit of extra code, but it's old code so I'd leave it the way it is.